src/ui/processing-animation.svelte

```diff
@@ -1,8 +1,22 @@
-<div class="processing"></div>
+<script lang="ts">
+  export let size: "small" | "medium" | "large" = "medium";
+
+  const sizeMap = {
+    small: { width: "8px", offset: "10px" },
+    medium: { width: "12px", offset: "20px" },
+    large: { width: "16px", offset: "30px" },
+  };
+</script>
+
+<div
+  class="processing"
+  style="--shadow-offset: {sizeMap[size].offset}; --width: {sizeMap[size]
+    .width};"
+></div>
 
 <style>
   .processing {
-    width: 12px;
+    width: var(--width, 12px);
```

Comment:
Mmmm, IIRC css variable names are not local, this could be set by someone outside this component. Also --width is quite generic so it's easy that it gets set.

Why do we need a var for this?

Reply:
> IIRC css variable names are not local
the CSS variable is being defined in the `style=` property, so they won't be accessible to the parents of this element, so no issue

> this could be set by someone outside this component.

In that case, it would be overriden by the `style` property, so no issue

>  Also --width is quite generic so it's easy that it gets set.

See above

> Why do we need a var for this?

To calculate, based on the size prop, both the widths and offsets of the dots with respect to each other dynamically, otherwise i'd have to create duplicate CSS classes and it would be much less maintainable imo.